lib/Frontend/Frontend.cpp

```diff
@@ -261,233 +262,300 @@ ModuleDecl *CompilerInstance::getMainModule() {
   return MainModule;
 }
 
-void CompilerInstance::performSema() {
-  const FrontendOptions &options = Invocation.getFrontendOptions();
-  const InputFileKind Kind = Invocation.getInputKind();
-  ModuleDecl *MainModule = getMainModule();
-  Context->LoadedModules[MainModule->getName()] = MainModule;
-
-  auto modImpKind = SourceFile::ImplicitModuleImportKind::Stdlib;
+static void
+addAdditionalInitialImportsTo(SourceFile *SF,
+                              ModuleDecl *objCModuleUnderlyingMixedFramework,
+                              ModuleDecl *importedHeaderModule,
+                              SmallVectorImpl<ModuleDecl *> &importModules) {
+  SharedTimer timer("performSema-addAdditionalInitialImportsTo");
+  if (!objCModuleUnderlyingMixedFramework && !importedHeaderModule &&
+      importModules.empty())
+    return;
 
-  if (Kind == InputFileKind::IFK_SIL) {
-    assert(BufferIDs.size() == 1);
-    assert(MainBufferID != NO_SUCH_BUFFER);
-    // Assume WMO, if a -primary-file option was not provided.
-    createSILModule(!options.PrimaryInput.hasValue());
-    modImpKind = SourceFile::ImplicitModuleImportKind::None;
-  } else if (Invocation.getParseStdlib()) {
-    modImpKind = SourceFile::ImplicitModuleImportKind::Builtin;
+  using ImportPair =
+      std::pair<ModuleDecl::ImportedModule, SourceFile::ImportOptions>;
+  SmallVector<ImportPair, 4> additionalImports;
+
+  if (objCModuleUnderlyingMixedFramework)
+    additionalImports.push_back(
+        {{/*accessPath=*/{}, objCModuleUnderlyingMixedFramework},
+         SourceFile::ImportFlags::Exported});
+  if (importedHeaderModule)
+    additionalImports.push_back({{/*accessPath=*/{}, importedHeaderModule},
+                                 SourceFile::ImportFlags::Exported});
+  if (!importModules.empty()) {
+    for (auto &importModule : importModules) {
+      additionalImports.push_back({{/*accessPath=*/{}, importModule}, {}});
+    }
   }
 
-  switch (modImpKind) {
+  SF->addImports(additionalImports);
+}
+
+// Implicitly import the SwiftOnoneSupport module in non-optimized
+// builds. This allows for use of popular specialized functions
+// from the standard library, which makes the non-optimized builds
+// execute much faster.
```

Comment:
Nitpick: may as well make this a doc comment.

Reply:
Done. Where can I go to read about doc comments? When is it best to employ one?

Reply:
PS: Is it sufficient to use three slashes, or is something else needed? (Or if I read up on those I guess I'll know.)

Reply:
Three slashes is fine. We use Doxygen format, following the [LLVM conventions](http://llvm.org/docs/CodingStandards.html#doxygen-use-in-documentation-comments).